app/controllers/api/users_controller.rb

```diff
@@ -2,98 +2,77 @@ class Api::UsersController < ApplicationController
+          }
+        }, status: :ok
+      else
+        head :forbidden
+      end
+    end
   end
 
-	def index
-    # TODO: I guess this is a placeholder for now, right?
-		i = {"user" => "Hello world!"}
-		render json: i
-	end
-
   def register
-    @user = User.create(user_params)
+    @user = User.new(user_params)
     if @user.save
-      response = { message: 'User created successfully' }
-      render json: {
-        message: response,
-        # TODO: What happens, if the AuthenticateUser call fails?
-        access_token: AuthenticateUser.new(user_params[:name], user_params[:password]).call
-      },
-      status: :created
+      if authenticator.authenticated
+        render json: {
+          user: {
+            id: @user.id,
+            name: @user.name,
+            access_token: authenticator.token
+          }
+        }, status: :created
+      else
+        render json: {errors: authenticator.errors}, status: :unprocessable_entity
+      end
     else
-      render json: @user.errors, status: :bad
+      render json: {errors: @user.errors}, status: :unprocessable_entity
     end
   end
 
-  # TODO: This method should be private
-  def user_params
-    params.permit(
-      :name,
-      :password
-    )
-  end
-
-  # TODO: Make a plan on how to connect users first, then implement accordingly.
-  # Discuss with Naveen
   def connect
-    @current_user.update(active: 1)
-    # TODO: Never do this kind of loop in a controller!
-    5.times do
-      if User.where(active: 1).count>1
-        @chatroom = Chatroom.create
-        @user1 = User.where(active: 1).order(:updated_at).first
-        @user2 = User.where(active: 1).order(:updated_at).second
-        ChatroomsUser.create(chatroom: @chatroom, user: @user1)
-        ChatroomsUser.create(chatroom: @chatroom, user: @user2)
-        @user1.update(active: 2)
-        @user2.update(active: 2)
-        if @user1 == @current_user or @user2 == @current_user
-          render json: {
-            chatroom_id: @chatroom.id,
-            message: "Chatroom created"
-          }
-        end
-        break
-      else
-        # TODO: Never use sleep() within a controller!
-        sleep(5)
-      end
+    @current_user.pending!
+    if User.where(connection_status: :pending).count > 1
+      @other = User.where(connection_status: :pending).where.not(id: @current_user.id).first
+      ConnectService.new(@current_user, @other).chat
     end
+    render json: {
+      connection: { 
+        status: "pending"
+      }
+    }, status: :ok
   end
 
-  # TODO: This method should be private
-  # Also, why don't you use authenticate_request of ApplicationController instead?
-  # That method also sets the current_user
-  def set_current_user
-    # TODO: What hapens if the request can not be authorized?
-    @current_user = AuthorizeApiRequest.new(request.headers).call
-  end
+  def disconnect
+    @current_user.inactive!
+    @current_user.update(active_chatroom: :null)
```

Comment:
You can basically merge those to lines in one by just calling @current_user.update(status: :inactive, active_chatroom: nil)
Also do not use symbol for null values
In Ruby, the null value is actually called nil